server/grpcservice/util.go

```diff
@@ -31,8 +31,12 @@ func okResponseHeader() *commonpb.ResponseHeader {
 
 func responseHeader(err error, errMsg string) *commonpb.ResponseHeader {
 	code, b := coderr.GetCauseCode(err)
-	if !b {
+	if !b && err != nil {
```

Comment:
And I guess the error passed to this function should be ensured not nil, so why should we do the check `err != nil`.

Another thing to fix:  rename the `b` to `ok`.